apollo-federation/src/query_plan/operation.rs

```diff
@@ -2873,10 +2877,18 @@ impl SelectionSet {
                 // in-place, we eagerly construct the selection that needs to be rebased on the target
                 // schema.
                 let element = OpPathElement::clone(ele);
-                let selection = Selection::from_element(
-                    element,
-                    selection_set.map(|set| SelectionSet::clone(set)),
-                )?;
+                let selection_set = selection_set.map(|selection_set| {
+                        selection_set.rebase_on(
+                            &element.sub_selection_type_position()?.ok_or_else(|| {
+                                FederationError::internal("unexpected: Element has a selection set with non-composite base type")
+                            })?,
+                            &NamedFragments::default(),
```

Comment:
Are we sure this is right? If `selection_set` contains references to named fragments, they need to exist in this map in order to rebase correctly. I'm not really sure what we *should* provide here. If we're not sure we should at least add a comment, and if we are sure we should mention that too i think.